app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuSitePermissionsCommonRobot.kt

```diff
@@ -0,0 +1,169 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui.robots
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.Visibility
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.CoreMatchers.allOf
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.assertIsChecked
+import org.mozilla.fenix.helpers.click
+
+/**
+ * Implementation of Robot Pattern for the settings Site Permissions sub menu.
+ */
+class SettingsSubMenuSitePermissionsCommonRobot {
+
+    fun verifyNavigationToolBarHeader(header: String) = assertNavigationToolBarHeader(header)
+
+    fun verifyBlockAudioAndVideoOnMobileDataOnlyAudioAndVideoWillPlayOnWiFi() = assertBlockAudioAndVideoOnMobileDataOnlyAudioAndVideoWillPlayOnWiFi()
+
+    fun verifyBlockAudioOnly() = assertBlockAudioOnly()
+
+    fun verifyVideoAndAudioBlockedRecommended() = assertVideoAndAudioBlockedRecommended()
+
+    fun verifyCheckAutoPlayRadioButtonDefault() = assertCheckAutoPayRadioButtonDefault()
+
+    fun verifyassertAskToAllowRecommended() = assertAskToAllowRecommended()
+
+    fun verifyassertBlocked() = assertBlocked()
+
+    fun verifyCheckRadioButtonDefault() = assertCheckRadioButtonDefault()
+
+    fun verifyBlockedByAndroid() = assertBlockedByAndroid()
+
+    fun verifyToAllowIt() = assertToAllowIt()
+
+    fun verifyGotoAndroidSettings() = assertGotoAndroidSettings()
+
+    fun verifyTapPermissions() = assertTapPermissions()
+
+    fun verifyToggleNameToON(name: String) = assertToggleNameToON(name)
+
+    fun verifyGoToSettingsButton() = assertGoToSettingsButton()
+
+    fun verifySitePermissionsAutoPlaySubMenuItems() {
+        verifyBlockAudioAndVideoOnMobileDataOnlyAudioAndVideoWillPlayOnWiFi()
+        verifyBlockAudioOnly()
+        verifyVideoAndAudioBlockedRecommended()
+        verifyCheckAutoPlayRadioButtonDefault()
+    }
+
+    fun verifySitePermissionsCameraSubMenuItems() {
+        verifyassertAskToAllowRecommended()

```

Comment:
This robot looks better now, less duplication of code. But I think we could go a step further reducing more duplication...for example, moving all the commons functions you have here and the in the other `verifySitePermissions...` to a common  `verifySitePermissionsCommonChecks (or the name you like)` since most of them are the same except for check...

Reply:
I didn't get you exactly padon for done. Could you please able to be specific ?

Reply:
Yeah ok I get it. I will do it.

Reply:
-corrections are made :)